include/zephyr/drivers/stepper_motor.h

```diff
@@ -0,0 +1,439 @@
+	uint16_t micro_step_resolution;
+};
+
+__subsystem struct stepper_motor_api {
+	stepper_motor_reset_t stepper_motor_reset;
+	stepper_motor_set_position_t stepper_motor_set_position;
+	stepper_motor_get_position_t stepper_motor_get_position;
+	stepper_motor_get_stall_status_t stepper_motor_get_stall_status;
+	stepper_motor_run_t stepper_motor_run;
+	stepper_motor_stop_t stepper_motor_stop;
+	stepper_motor_calibrate_t stepper_motor_calibrate;
+	stepper_motor_register_calibrate_func_t stepper_motor_register_calibrate_func;
+
+	stepper_motor_controller_reg_write_t stepper_motor_controller_write_reg;
+	stepper_motor_controller_reg_read_t stepper_motor_controller_read_reg;
+};
+
+/**
+ * @brief Reset stepper motor device
+ *
+ * @param dev Pointer to the stepper motor device
+ * @param motor_index index of the motor to be controlled
+ *
+ * @return 0 if successful, negative code if failure
+ */
+__syscall int stepper_motor_reset(const struct device *dev, uint8_t motor_index);
+
+static inline int32_t z_impl_stepper_motor_reset(const struct device *dev, uint8_t motor_index)
+{
+	const struct stepper_motor_api *api = (const struct stepper_motor_api *)dev->api;
+
+	return api->stepper_motor_reset(dev, motor_index);
+}
+
+/**
+ * @brief Run a.k.a freewheel a stepper motor device
+ *
+ * @param dev Pointer to the stepper motor device
+ * @param motor_index index of the motor to be controlled
+ * @param run_info Pointer to the run information such as direction and velocity
+ *
+ * @return 0 if successful, negative code if failure
+ */
+__syscall int32_t stepper_motor_run(const struct device *dev, uint8_t motor_index,
+				    const struct stepper_motor_run_info *run_info);
+
+static inline int32_t z_impl_stepper_motor_run(const struct device *dev, uint8_t motor_index,
+					       const struct stepper_motor_run_info *run_info)
+{
+	const struct stepper_motor_api *api = (const struct stepper_motor_api *)dev->api;
+
+	return api->stepper_motor_run(dev, motor_index, run_info);
+}
+
+/**
+ * @brief Stop a stepper motor device
+ *
+ * @param dev Pointer to the stepper motor device
+ * @param motor_index index of the motor to be controlled
+ *
+ * @return 0 if successful, negative code if failure
+ */
+__syscall int32_t stepper_motor_stop(const struct device *dev, uint8_t motor_index);
+
+static inline int32_t z_impl_stepper_motor_stop(const struct device *dev, uint8_t motor_index)
+{
+	const struct stepper_motor_api *api = (const struct stepper_motor_api *)dev->api;
+
+	return api->stepper_motor_stop(dev, motor_index);
+}
+
+/**
+ * @brief Set position of a stepper motor device
+ *
+ * @param dev Pointer to the stepper motor device
+ * @param motor_index index of the motor to be controlled
+ * @param position_info Pointer to position information such as format, type and position
+ *
+ * @return 0 if successful, negative code if failure
+ */
+__syscall int32_t
+stepper_motor_set_position(const struct device *dev, uint8_t motor_index,
```

Comment:
The newline character is not getting parsed correctly, hence gen_syscalls.py throws up the following exception `raise SyscallParseException("Malformed system call invocation")`

Reply:
```
__syscall int32_t stepper_motor_set_position(const struct device *dev, uint8_t motor_index,
					const struct stepper_motor_position_info *position_info);
```